### DISCUSSION TITLE
Blessed shall be the ARM (poor)

### DISCUSSION THREAD

**Discussion by clort81** [Try/OpenGothic#166] [General]
To avoid spamming 'Issues', I am starting a discussion for testers/users of OpenGothic on ARM boards. 
Some low-cost boards may be capable of running OpenGothic:

- Raspberry Pi4 (Broadcom Vulkan drivers)
- Nvidia Jetson Nano (nvidia closed drivers)
- Khadas Vim3 Pro (panfrost open-source Vulkan driver is WIP)
- Odroid N2/N2+ (panfrost open-source Vulkan driver is WIP)

I managed to build OpenGothic yesterday by switching my gcc from 10 to 9
```
631904 Jul 11 07:19 /usr/local/bin/Gothic2Notr 
```

Not having a windows box any more, I found no way to extract the game data from Gothic2.  The data seems to be in 
```
632312 Gothic2-Setup.exe
632312 Gothic2-Setup.W02
372288 Gothic2-Setup.W03
```
**Can anyone help me figure out how to extract these without setting up a Windblows machine again?**

From my Gothic 1 CDs, I could extract datafiles with 'unshield' utility:
```
mkdir gothic1
unshield -d gothic1 x data1.cab
unshield -d gothic1 x data1.cab
```
However Gothic2Notr complained about not finding game files
```
oldgames$ Gothic2Notr -g /oldgames/gothic1/
invalid gothic path: "/oldgames/gothic1/"

---crashlog(std::logic_error(gothic not found!))---
GPU: ?
#1: b - /lib/aarch64-linux-gnu/libstdc++.so.6(+0x9edb0) [0xffff86f20db0]
#2: b - /lib/aarch64-linux-gnu/libstdc++.so.6(+0x9f0a4) [0xffff86f210a4]
#3: Gothic::Gothic(int, char const**) - Gothic2Notr(_ZN6GothicC1EiPPKc+0xcfc) [0x5315c0]
#4: main - Gothic2Notr(main+0x40) [0x57d574]

Aborted
```
The contents of /oldgames/gothic1 are:
```
721968  /oldgames/gothic1/App_Executables
32      /oldgames/gothic1/_Engine_Engine_Files
4       /oldgames/gothic1/_Engine_Kernel_Placeholder
224     /oldgames/gothic1/_Engine_ScriptEngine
284     /oldgames/gothic1/_Engine_SelfRegistering
216     /oldgames/gothic1/_Support_English_Files
8       /oldgames/gothic1/_Support_English_String_Tables
33888   /oldgames/gothic1/_Support_Language_Independent_OS_Independent_Files
324     /oldgames/gothic1/_Support_Non-SelfRegistering
```
Running strace I see Gothic2Notr looks for 
```
openat(AT_FDCWD, "/oldgames/gothic1/_work/", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
```
The _work is under App_Executables in my extracted directory tree, so I re-invoke with:
```
$ Gothic2Notr -g /oldgames/gothic1/App_Executables
```
And it finds data files!  Much things appearing in console! \o/

However it segfaults with:
```
no *.ini file in path - using default settings

---crashlog(SIGSEGV)---
GPU: ?
#1:  - /lib/aarch64-linux-gnu/libvulkan.so.1(+0xcf18) [0xffff990ebf18]
#2:  - /lib/aarch64-linux-gnu/libvulkan.so.1(+0x23290) [0xffff99102290]
#3: vkCreateInstance - /lib/aarch64-linux-gnu/libvulkan.so.1(vkCreateInstance+0x314) [0xffff99104ff4]
#4: Tempest::Detail::VulkanInstance::VulkanInstance(bool) - /usr/local/lib/libTempest.so(_ZN7Tempest6Detail14VulkanInstanceC2Eb+0x150) [0xffff996a285c]
#5: Tempest::VulkanApi::VulkanApi(Tempest::ApiFlags) - /usr/local/lib/libTempest.so(_ZN7Tempest9VulkanApiC2ENS_8ApiFlagsE+0x3c) [0xffff996a394c]
#6: mkApi(Gothic&) - Gothic2Notr(_Z5mkApiR6Gothic+0x58) [0x57d508]
#7: main - Gothic2Notr(main+0x50) [0x57d584]

Segmentation fault
```
This is expected since panfrost drivers are just beginning to support Vulkan.  We can hope to see them reach working Vulkan 1.0 though. :)

I hope that documenting my progress here may help someone.

**Comment by luziferius** [comment]
> Can anyone help me figure out how to extract these without setting up a Windblows machine again?

For Gothic 1, I found that the disc installer works fine with wine. The process may hang at the end of the installation process, never showing the finished screen. It did this for me in about 9 out of 10 tries. But the installation works nonetheless, after killing the installer process. I think it hangs due to the mid-installation disk change. If you create images and insert them into two different virtual drives, it should finish.

I _assume_ the G2 disks should work similarly.

So install wine, insert your game disks (or use cdemu to mount images), install the game into a new wine prefix like 

    WINEPREFIX=$HOME/G2_TEMP /path/to/setup.exe

Then extract the game files and remove the created wine prefix.

**Reply by clort81** [reply]
> > Can anyone help me figure out how to extract these without setting up a Windblows machine again?
> 
> For Gothic 1, I found that the disc installer works fine with wine. The process may hang at the end of the installation process, never showing the finished screen. It did this for me in about 9 out of 10 tries. But the installation works nonetheless, after killing the installer process. I think it hangs due to the mid-installation disk change. If you create images and insert them into two different virtual drives, it should finish.
> 
> I _assume_ the G2 disks should work similarly.
> 
> So install wine, insert your game disks (or use cdemu to mount images), install the game into a new wine prefix like
> 
> ```
> WINEPREFIX=$HOME/G2_TEMP /path/to/setup.exe
> ```
> 
> Then extract the game files and remove the created wine prefix.

Thanks!  got gothic2 installed with box86+wine !

It's rather fun!  Would like more performance, so I hope panfrost gets working vulkan sometime next year :)
